core/api/views/business_plan.py

```diff
@@ -126,3 +133,136 @@ def export(self, *args, **kwargs):
+                    description="Comment text",
+                ),
+            },
+        ),
+    )
+    def _comments_update_or_create(self, request, *args, **kwargs):
+        business_plan = self.get_object()
+        comment_type = request.data.get("comment_type")
+        comment = request.data.get("comment", "")
+
+        user = request.user
+        if comment_type == "comment_agency":
+            if user.user_type != user.UserType.AGENCY:
+                return Response(
+                    {comment_type: f"Invalid value {comment}"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            business_plan.comment_agency = comment
+
+        if comment_type == "comment_secretariat":
+            if user.user_type != user.UserType.SECRETARIAT:
+                return Response(
+                    {comment_type: f"Invalid value {comment}"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            business_plan.comment_secretariat = comment
+
+        business_plan.save()
+        serializer = self.get_serializer(business_plan)
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def post(self, request, *args, **kwargs):
+        return self._comments_update_or_create(request, *args, **kwargs)
+
+    def put(self, request, *args, **kwargs):
+        return self._comments_update_or_create(request, *args, **kwargs)
+
+
+class BPFileView(generics.GenericAPIView):
+    """
+    API endpoint that allows uploading business plan file.
+    """
+
+    queryset = BusinessPlan.objects.all()
+    serializer_class = BPFileSerializer
+    lookup_field = "id"
+
+    ACCEPTED_EXTENSIONS = [
+        ".pdf",
+        ".doc",
+        ".docx",
+        ".xls",
+        ".xlsx",
+        ".csv",
+        ".ppt",
+        ".pptx",
+        ".jpg",
+        ".jpeg",
+        ".png",
+        ".gif",
+        ".zip",
+        ".rar",
+        ".7z",
+    ]
+
+    def _file_create(self, request, *args, **kwargs):
+        business_plan = self.get_object()
+
+        files = request.FILES
+        if not files:
+            return Response(
+                {"feedback_file": "File not provided"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        filename, file = next(files.items())
+        extension = os.path.splitext(filename)[-1]
+        if extension not in self.ACCEPTED_EXTENSIONS:
+            return Response(
+                {"feedback_file": f"File extension {extension} is not valid"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        business_plan.feedback_filename = filename
+        business_plan.feedback_file = file
+        business_plan.save()
+        serializer = self.get_serializer(business_plan)
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def post(self, request, *args, **kwargs):
+        return self._file_create(request, *args, **kwargs)
```

Comment:
I think you should check if the business plan already has a file. If it does, then you should delete the old file and upload the new one.